pkg/virt-launcher/virtwrap/converter/vcpu/vcpu.go

```diff
@@ -424,11 +424,10 @@ func FormatEmulatorThreadPin(cpuPool VCPUPool, CPUManagerPolicyBetaOption v12.CP
 	if fullpCPUsOnly {
 		availableThread, err = cpuPool.FitThread()
 		if err != nil {
-			e := fmt.Errorf("no second CPU allocated for the emulation thread: %v", err)
-			log.Log.Reason(e).Error("failed to format emulation thread pin")
-			return "", e
+			log.Log.Info("second emulation thread not allocated (best effort)")
```

Comment:
I don't know why this is a best effort allocation. 
Wouldn't it be better to check whether we need to do this? (Whether the number of thread is ever or odd)

Reply:
short answer: we can't know - counting the amount of CPUs on the CPUSet will always be even now.

longer answer: The part where we set the +1/+2 CPUs occurs on the virt-controller. 
By the time we reach virt-launcher - we already have an even amount of CPUs (because it was already aligned to be even in on virt-controller's [WithCPUPinning()](https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-controller/services/renderresources.go#L216)

So in order to "know" if we want to ask for a second thread for the emulatorThreadPin, we can do one of these options:
1. We still have the VMI object so we can redo the logic we do on virt-controller, i.e. [WithCPUPinning()](https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-controller/services/renderresources.go#L216).
2. ,We can rely that virt-controller added +1/+2 - if we have 2 extra CPUs - then take them, otherwise only take 1.. This why I called it "best effort".. though maybe the phrasing is off-putting?..

I am pf course open for other options :) WDYT?

Reply:
On second thought - I think extracting the logic from [WithCPUPinning()](https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-controller/services/renderresources.go#L216) is doable, and should be stable enough.